CapybaraLearning/spec/menu_spec.rb

```diff
@@ -1,185 +1,36 @@
-    expect(@header_page).to have_current_path @header_page.header_page_path
-  end
-
-  it 'Verify navigating to previous page' do
-    @header_page.go_back
-    expect(@appearance_page).to have_current_path @appearance_page.appearance_page_path
-  end
-
-  it 'Verify navigating to Plugins page' do
-    @plugins_page = @appearance_page.plugins_button_click
-    expect(@plugins_page).to have_content @plugins_page.plugins_page_message
-  end
-
-  it 'Verify navigating to User page' do
-    @user_page = @plugins_page.users_button_click
-    expect(@user_page).to have_current_path @user_page.users_page_path
-  end
-
-  it 'Verify navigating to Your Profile page' do
-    @profile_page = @user_page.your_profile_button_click
-    expect(@profile_page).to have_content @profile_page.your_profile_page_message
-  end
-
-  it 'Verify navigating to Tools page' do
-    @tools_page = @profile_page.tools_button_click
-    expect(@tools_page).to have_content @tools_page.tools_page_message
-  end
-
-  it 'Verify navigating to Import page' do
-    @import_page = @tools_page.import_button_click
-    expect(@import_page).to have_content @import_page.import_page_message
-  end
-
-  it 'Verify navigating to Export page' do
-    @export_page = @import_page.export_button_click
-    expect(@export_page).to have_content @export_page.export_page_message
-  end
-
-  it 'Verify navigating to Site Health page' do
-    @site_health_page = @export_page.site_health_button_click
-    expect(@site_health_page).to have_content @site_health_page.site_health_page_message
-  end
-
-  it 'Verify navigating to Site General Setting page' do
-    @settings_general_page = @site_health_page.settings_general_button_click
-    expect(@settings_general_page).to have_content @settings_general_page.settings_general_page_message
-  end
-
-  it 'Verify navigating to Site Writing page' do
-    @writing_page = @settings_general_page.writing_button_click
-    expect(@writing_page).to have_content @writing_page.writing_page_message
-  end
-
-  it 'Verify navigating to Site Reading page' do
-    @reading_page = @writing_page.reading_button_click
-    expect(@reading_page).to have_content @reading_page.reading_page_message
-  end
-
-  it 'Verify navigating to Discussion page' do
-    @discussion_page = @reading_page.discussion_button_click
-    expect(@discussion_page).to have_content @discussion_page.discussion_page_message
-  end
-
-  it 'Verify navigating to Permalink page' do
-    @permalink_page = @discussion_page.permalink_button_click
-    expect(@permalink_page).to have_content @permalink_page.permalink_page_message
-  end
-
-  it 'Verify navigating to Privacy page' do
-    @privacy_page = @permalink_page.privacy_button_click
-    expect(@privacy_page).to have_content @privacy_page.privacy_page_message
+  it 'Verify navigation to all pages' do
+    pages_array.each do |current_page|
+      current_page.visit_page
+      expect(current_page).to have_content current_page.page_message
```

Comment:
In general it is not a good approach to verify that you are on a page by just checking some text, since for large heave applications, this text might appear right away, but page is not yet loaded. So it's better, to identify the object (table, chart) that is loaded last and check if it is there. But for this training task and taking into account demo site that is used, you can leave current approach with page_message.